telluric/georaster.py

```diff
@@ -101,21 +103,97 @@ def merge_all(rasters, roi=None, dest_resolution=None, merge_strategy=MergeStrat
        When roi is provided the ul_corner, shape and crs are ignored
     """
     if dest_resolution is None:
-        dest_resolution = rasters[0].resolution()
+        dest_resolution = rasters[0].res_xy()
 
     # Create empty raster
     empty = GeoRaster2.empty_from_roi(
         roi, resolution=dest_resolution, band_names=rasters[0].band_names,
         dtype=rasters[0].dtype, shape=shape, ul_corner=ul_corner, crs=crs)
 
-    projected_rasters = [_prepare_other_raster(empty, raster) for raster in rasters]
-    projected_rasters = [raster for raster in projected_rasters if raster is not None]
-    func = partial(_merge, merge_strategy=merge_strategy)  # type: Callable[[_Raster, _Raster], _Raster]
-    raster = reduce(func, projected_rasters, empty)
-    return empty.copy_with(image=raster.image, band_names=raster.band_names)
+    # Create a list of single band rasters
+    all_band_names, projected_rasters = _prepare_rasters(rasters, merge_strategy, empty)
+
+    if all_band_names:
```

Comment:
correct me if I am wrong you don't use the list of bands anywhere in the if code block, does that make sense? if so why did you calculate the list of bands in the first place?

Reply:
It's computed internally in `_prepare_rasters` to explode only the required bands, and it's returned here to raise a proper error message.